submissions/chimap/components/map/global.map.js

```diff
@@ -4,28 +4,33 @@ import useInfo from '../../hooks/getInfo'
 import countryData from '../../utils/countries.json'
 import ServiceCard from "../card/service.card";
 import wc from 'which-country'
-import { filterByCountry, parseAssetType } from '../../helpers/query'
+import { filterByCountry,filterGiftCardByCountry, parseAssetType } from '../../helpers/query'
 import CountryList from "country-list-with-dial-code-and-flag"
 
 
 
 const GlobalMap = (props) => {
   const [showPopup, setShowPopup] = useState(false)
   const [lngLat, setlngLng] = useState({ lng: 0, lat: 0 })
-  const [darkMode,setDarkMode]=useState(true)
+  const [darkMode, setDarkMode] = useState(true)
 
   const countryList = countryData
 
   const [country, setCountry] = useState("")
   const [filteredData, setFilteredData] = useState([])
   const [filteredType, setFilteredType] = useState([])
+  const [filteredMobileMoney,setFilteredMobileMoney]=useState([])
+  // const [filteredGiftCard,setFilteredGiftCard]=useState([])
 
   // hook to call api
   const apiData = useInfo()
```

Comment:
Consider making the useInfo hook to take `country` or default to US. You might have to reconsider if you want to use a hook since the API needs to be called everytime a country is selected

And everytime you click on a country, call the API to fetch the info with the countryCode as outlined at https://chimoney.readme.io/reference/get_v0-2-info-assets

So what this does is that it gives you a different list of Gift Cards and every country has Gift cards unique to them. The current implementation always only uses US Gift cards

Reply:
@uchibeke this endpoint [https://chimoney.readme.io/reference/get_v0-2-info-assets](https://chimoney.readme.io/reference/get_v0-2-info-assets) returns all the info which is saved during the session, I wrote an helper to query this info so as to avoid repeated API call for each clicks.
 Thought this may sound more optimized that having to call it on every clicks on the map.

The results are same for both.
Let me know what you think?